### PR TITLE
Schema updates, miss, and improvements

### PR DESCRIPTION
**Pull request by trasher**
Some changes on inventory schema including:
- List instances under related database (makes updates easier)
- Do not validate JSON that contains non existing nodes,
- Add fields used glpi side but not present in schema 
- ...

### CODE REVIEW

**Review by g-bougard — APPROVED** [review]

File: lib/php/Converter.php [hunk 1]
```diff
@@ -500,11 +500,20 @@ private function convertTo01(array $data)
...
         if (isset($data['content']['antivirus'])) {
             foreach ($data['content']['antivirus'] as &$av) {
+                //expiration date format
                 if (($convertedDate = $this->convertDate($av['expiration'] ?? '')) !== null) {
                     $av['expiration'] = $convertedDate;
                 } else {
                     unset($av['expiration']);
                 }
+
+                //old proeperty
```
Comment (g-bougard) [inline lib/php/Converter.php:510]: Little typo in comment

**Review by g-bougard — CHANGES_REQUESTED** [review]
"port" is still on "required" properties for "databases" but it has been moved to "instances" members

**Review by g-bougard — APPROVED** [review]

**Review by g-bougard — COMMENTED** [review]
I see you added a commit "More fixes" which adds a lot of changes in inventory schema. I don't understand what are these schema "fixes" without explanation. It seems even to me you may introduce back specs that was removed in the original XML format.

Finally, the PR title has less and less purpose with the PR content.

**Comment by trasher** [comment]
This PR is still a draft

File: inventory.schema.json [hunk 2]
```diff
@@ -54,7 +54,8 @@
             "userid": {
               "type": "string"
             }
-          }
+          },
+          "additionalProperties": false
```
Comment (trasher) [inline inventory.schema.json:58]: I've added this on all `content` properties but `cartridges` that can contains several values.

With this, validation will fail if JSON contains a property not listed in the schema. From the inventories I have locally (20 or 30K inventory files); I had to drop some, and I can convert some others.

File: inventory.schema.json [hunk 3]
```diff
@@ -2294,7 +2333,7 @@
               "orientation": {
                 "type": "string"
               },
-              "focallenght": {
+              "focallength": {
```
Comment (trasher) [inline inventory.schema.json:2336]: Typo

File: inventory.schema.json [hunk 4]
```diff
@@ -2351,6 +2391,9 @@
                     "type": "string",
                     "title": "Remote device port description, from CDP/LLDP"
                   },
+                  "ifalias": {
```
Comment (trasher) [inline inventory.schema.json:2394]: Existing, and used

File: inventory.schema.json [hunk 5]
```diff
@@ -2379,6 +2422,9 @@
               "ifdescr": {
                 "type": "string"
               },
+              "ifalias": {
```
Comment (trasher) [inline inventory.schema.json:2425]: Existing, and used

File: inventory.schema.json [hunk 6]
```diff
@@ -2442,6 +2488,26 @@
                   "00:00:00:00:00:00"
                 ]
               },
+              "ips": {
```
Comment (trasher) [inline inventory.schema.json:2491]: Existing and used

File: inventory.schema.json [hunk 7]
```diff
@@ -2489,8 +2556,15 @@
               },
               "name": {
                 "type": "string"
+              },
+              "serial": {
```
Comment (trasher) [inline inventory.schema.json:2560]: Existing and used

File: inventory.schema.json [hunk 8]
```diff
@@ -2803,19 +2871,54 @@
...
-              "last_boot_date": {
-                "examples": [
-                  "2021-06-03 10:16:47"
-                ],
-                "$ref": "#/definitions/datetime"
-              },
               "manufacturer": {
                 "type": "string"
+              },
+              "instances": {
```
Comment (trasher) [inline inventory.schema.json:2877]: Use a subproperty fo instances, remove instance only data from database

**Review by cedric-anne — APPROVED** [review]

File: inventory.schema.json (hunk 6 above)
Comment (g-bougard) [reply inventory.schema.json:2491]: Agree too

File: inventory.schema.json (hunk 7 above)
Comment (g-bougard) [reply inventory.schema.json:2560]: Agree too

File: inventory.schema.json (hunk 8 above)
Comment (g-bougard) [reply inventory.schema.json:2877]: "is_active" & "is_onbackup" properties are also still set on "databases" object. So why keeping them on "databases" object ?
Also about "creation_date" & "update_date", why are they kept on databases object ? I think they would make more sense on "instances" objects.

**Review by orthagh — APPROVED** [review]

File: inventory.schema.json (hunk 4 above)
Comment (trasher) [reply inventory.schema.json]: Indeed, this is not existing in connections; I've removed it.

**Pull request merged**